docs/documentation/basics/batching-simulations.md

```diff
@@ -1,86 +1,255 @@
 ```
 
+```text title="Output"
+Shape of result.states: (3, 4, 11, 2, 1)
 ```
-==== SEResult ====
-Solver  : Tsit5
-States  : Array complex64 (4, 11, 2, 1) | 0.69 Kb
-Expects : Array complex64 (4, 1, 11) | 0.34 Kb
-Infos   : avg. 6.0 steps (6.0 accepted, 0.0 rejected) | infos shape (4,)
-```
 
-The returned `states` array has shape `(4, 11, 2, 1)` where `4` is the number of initial states, `11` is the number of saved states (the length of `tsave`) and `(2, 1)` is the shape of a single state.
+The returned states is an array with shape _(3, 4, 11, 2, 1)_, where _3_ is the number of Hamiltonians, _4_ is the number of initial states, _11_ is the number of saved states, and _(2, 1)_ is the shape of a single state.
+
+!!! Note
+    All relevant `result` attributes are batched. For example if you specified `exp_ops`, the resulting expectation values `result.expects` will be an array with shape _(3, 4, len(exp_ops), 11)_.
+
+Importantly, **batched simulations are not run sequentially in a `for` loop**. What is meant by *batching* is that instead of evolving from initial to final time a single state with shape _(2, 1)_ for each combination of argument, the whole batched state _(3, 4, 2, 1)_ is evolved **once** from initial to final time, which is much more efficient.
+
+## Batching modes
+
+There are two ways to batch simulations in dynamiqs: **cartesian batching** and **flat batching**.
+
+### Cartesian batching
+
+The simulation runs for all possible combinations of Hamiltonians, jump operators and initial states. This is the default mode.
+
+=== "`dq.sesolve`"
+    For `dq.sesolve`, the returned array has shape:
+    ```
+    result.states.shape = (...H, ...psi0, ntsave, n, 1)
+    ```
+    where `...x` indicates the batching shape of `x`, i.e. its shape without the last two dimensions.
+
+    !!! Example "Example: Cartesian batching with `dq.sesolve`"
+        - `H` has shape _(2, 3, n, n)_,
+        - `psi0` has shape _(4, n, 1)_,
+
+        then `result.states` has shape _(2, 3, 4, ntsave, n, 1)_.
+
+=== "`dq.mesolve`"
+    For `dq.mesolve`, the returned array has shape:
+    ```
+    result.states.shape = (...H, ...L0, ...L1,  (...), ...rho0, ntsave, n, n)
+    ```
+    where `...x` indicates the batching shape of `x`, i.e. its shape without the last two dimensions.
+
+    !!! Example "Example: Cartesian batching with `dq.mesolve`"
+        - `H` has shape _(2, 3, n, n)_,
+        - `jump_ops = [L0, L1]` has shape _[(4, 5, n, n), (6, n, n)]_,
+        - `rho0` has shape _(7, n, n)_,
+
+        then `result.states` has shape _(2, 3, 4, 5, 6, 7, ntsave, n, n)_.
+
+### Flat batching
+
+The simulation runs for each set of Hamiltonians, jump operators and initial states using broadcasting. This mode can be activated by setting `cartesian_batching=False` in [`dq.Options`][dynamiqs.Options]. In particular for [`dq.mesolve()`][dynamiqs.mesolve], each jump operator can be batched independently from the others.
```

Comment:
The "In particular ..." is not specific to flat batching, we should move it elsewhere @gautierronan. I merge bc I don't want to block longer, this is already so much better than the previous version.